### PR TITLE
Fix package grouping for NIO examples and pubsub's autogenerated layer

### PR DESCRIPTION
**Pull request by mziccard**
The visual result is the following (compare it with the [current site](http://googlecloudplatform.github.io/gcloud-java/0.2.5/apidocs/index.html)).

screen shot 2016-07-04 at 9 24 36 am

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -384,11 +384,11 @@
                   </group>
                   <group>
                     <title>Example packages</title>
-                    <packages>com.google.cloud.examples*</packages>
+                    <packages>com.google.cloud.examples*:com.google.cloud.nio.examples</packages>
```
Comment (lesv) [inline pom.xml]: Should it be examples.*?  It's ok if it should be, examples* just offends my eye if it's not required.

File: pom.xml [hunk 2]
```diff
@@ -384,11 +384,11 @@
...
                   </group>
                   <group>
                     <title>SPI packages</title>
-                    <packages>com.google.cloud.spi:com.google.cloud.bigquery.spi:com.google.cloud.compute.spi:com.google.cloud.datastore.spi:com.google.cloud.dns.spi:com.google.cloud.pubsub.spi:com.google.cloud.resourcemanager.spi:com.google.cloud.storage.spi</packages>
+                    <packages>com.google.cloud.spi:com.google.cloud.bigquery.spi:com.google.cloud.compute.spi:com.google.cloud.datastore.spi:com.google.cloud.dns.spi:com.google.cloud.pubsub.spi*:com.google.cloud.resourcemanager.spi:com.google.cloud.storage.spi</packages>
```
Comment (lesv) [inline pom.xml]: spi* or spi.* ?

**Comment by lesv** [comment]
LGTM - if you could change to spi.* and examples.*, I'd be happier, but it's a personal bias, not a requirement.

**Comment by mziccard** [comment]
@lesv Fixed according to your comments!

**Comment by lesv** [comment]
Thanks.

**Pull request merged**